### PR TITLE
Support for hand tracking in openxr backend

### PR DESCRIPTION
**Pull request by Manishearth**
This can't be landed until we have support for microsoft extensions in openxrs (I have some experimentation in https://github.com/Ralith/openxrs/pull/24)

This mostly exists so that folks can play around with it

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @jdm 

This is using the xr-msft branch on https://github.com/servo/openxrs/

File: webxr/openxr/mod.rs [hunk 1]
```diff
@@ -121,14 +121,18 @@ impl OpenXrDiscovery {
...
-pub fn create_instance() -> Result<Instance, String> {
-    let entry = Entry::load().map_err(|e| format!("Entry::load {:?}", e))?;
+struct CreatedInstance {
+    instance: Instance,
+    supports_hands: bool,
+}
 
-    let extensions = entry
+fn create_instance(needs_hands: bool) -> Result<CreatedInstance, String> {
+    let entry = Entry::load().map_err(|e| format!("{:?}", e))?;
```
Comment (jdm) [inline webxr/openxr/mod.rs:129]: Keep the `Entry::load ` prefix in this message please.

File: webxr/openxr/mod.rs [hunk 2]
```diff
@@ -121,14 +121,18 @@ impl OpenXrDiscovery {
...
+    let supported = entry
         .enumerate_extensions()
-        .map_err(|e| format!("Entry::enumerate_extensions {:?}", e))?;
-    warn!("Available extensions:\n{:?}", extensions);
-
+        .map_err(|e| format!("{:?}", e))?;
```
Comment (jdm) [inline webxr/openxr/mod.rs:132]: Keep the `Entry::enumerate_extensions ` prefix.

File: webxr/openxr/input.rs [hunk 3]
```diff
@@ -147,18 +153,104 @@ impl OpenXRInput {
             )
             .unwrap();
 
-        let hand_tracker = if needs_hands {
+        let (_hand_tracker, joints) = if needs_hands {
```
Comment (jdm) [inline webxr/openxr/input.rs:156]: What's the purpose of ignoring this value?

File: webxr/openxr/input.rs [hunk 4]
```diff
@@ -171,14 +263,31 @@ impl OpenXRInput {
             click_state: ClickState::Done,
             squeeze_state: ClickState::Done,
             menu_gesture_sustain: 0,
-            hand_tracker,
+            _hand_tracker,
```
Comment (jdm) [inline webxr/openxr/input.rs:266]: Looks like this should be renamed?

File: webxr/openxr/input.rs [hunk 5]
```diff
@@ -306,13 +415,26 @@ impl OpenXRInput {
...
+                Some(Box::new(joints.map(|j, _| {
+                    j.as_ref()
+                        .and_then(|j| joint_for(j, frame_state, base_space))
+                })))
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```
Comment (jdm) [inline webxr/openxr/input.rs:429]: I had a hard time reading this snippet for some reason. Perhaps:
```rust
let hand = target_ray_origin
    .and_then(|_origin| self.joints.as_ref())
    .map(|joints| Box::new(joints.map(|j, _| {
        j.as_ref().and_then(|j| joint_for(j, frame_state, base_space))
    })));
```

File: webxr/openxr/input.rs (hunk 3 above)
Comment (Manishearth) [reply webxr/openxr/input.rs:156]: We need to hold on to it in the struct but it generates an unused error. I'll `allow()` it

**Comment by Manishearth** [comment]
Addressed

**Pull request merged**